share/revng/test/configuration/revng-qa/for-detect-abi.yml

```diff
@@ -65,6 +65,57 @@ tags:
         - -DSCALAR_SIZE_LIMIT=2
         - -DSTRUCT_SIZE_LIMIT=0
 
+  - name: i386-cdecl-microsoft-abi
+    implies: [i386]
+    variables:
+      ABI_NAME: Microsoft_x86_cdecl
+      CFLAGS:
+        - -DABIDEF='__attribute__((ms_abi, cdecl))'
+        - -DSCALAR_SIZE_LIMIT=2
+        - -DSTRUCT_SIZE_LIMIT=0
+
+  - name: i386-fastcall-microsoft-abi
+    implies: [i386]
+    variables:
+      ABI_NAME: Microsoft_x86_fastcall
+      CFLAGS:
+        - -DABIDEF='__attribute__((ms_abi, fastcall))'
+        - -DSCALAR_SIZE_LIMIT=2
+        - -DSTRUCT_SIZE_LIMIT=0
```

Comment:
1. These are rather obscure and need to be documented somewhere. At least at the top of this file.
2. Not necessarily for now, but we should consider whether we want to move all of this "configuration" into a `.h` file.
   i.e., Just `-DABI_Microsoft_x86_fastcall` and then in a `.h` file have:
   ```c
   #ifdef ABI_Microsoft_x86_fastcall
   #define ABIDEF __attribute__((ms_abi, fastcall))
   #define SCALAR_SIZE_LIMIT 2
   #define STRUCT_SIZE_LIMIT 0
   #elif ...
   ```
   These configurations are not "test configuration" but more like "ABI properties definitions", and I feel like they belong somewhere else.

Reply:
In a perfect world, we'd have a script to read our yaml definitions and figure these out from it. But currently that would cause a dependency cycle (since yamls are in revng and we _really_ don't want these ugly generators down there) - so I opted out to temporarily hand-write them until we come up with a better solution.

Reply:
We don't want to depend on revng.
I'd go for the header.

Reply:
I guess header is still an improvement compared to this. Even if not ideal.

Reply:
The header is ready, please take a look: `share/revng/test/tests/abi/abi.h`.